dot-gen/src/main/java/com/wipro/ats/bdre/ldg/GetDotForTable.java

```diff
@@ -0,0 +1,125 @@
+        for (LineageNode lineageNode : lineageNodeList) {
+            //this only checks the case where the node is src
+            LOGGER.info("------------------------inside");
+            lineageRelationList = getNodeIdForLineageRelation.execute(lineageNode.getNodeId());
+            for(LineageRelation lineageRelation:lineageRelationList) {
+                LOGGER.info("------------------------inside src check");
+                relationDot.append("\n" + lineageRelation.getDotString() + "\n");                                            //search for target table name in the DOT string
+                LOGGER.info("------------------" + relationDot);
+                targetTableNode.append("\n" + getLineageNodeByColName.getTableDotFromNodeId(lineageRelation.getLineageNodeByTargetNodeId(), srcTableNodeList.get(0)) + "\n");
+                LOGGER.info("-------------------------" + targetTableNode);
+                if("same-nodes".equals(targetTableNode)) {
+                    targetTableNode.append("\n" + getLineageNodeByColName.getTableDotFromNodeId(lineageRelation.getLineageNodeBySrcNodeId(), srcTableNodeList.get(0)) + "\n");
+                }
+                LOGGER.info("------------------------------relation DOT: \n" + relationDot);
+             }
+
+            //this should check whether the nodes are target -- ??
+            lineageRelationListForTargetNodes = getNodeIdForLineageRelation.getWhenTargetNode(lineageNode.getNodeId());
+            for(LineageRelation lineageRelation:lineageRelationListForTargetNodes) {
+                LOGGER.info("------------------------inside target check");
+                relationDot.append("\n" + lineageRelation.getDotString() + "\n");                                            //search for target table name in the DOT string
+                LOGGER.info("------------------" + relationDot);
+                targetTableNode.append("\n" + getLineageNodeByColName.getTableDotFromNodeId(lineageRelation.getLineageNodeBySrcNodeId(), srcTableNodeList.get(0)) + "\n");
+                LOGGER.info("-------------------------" + targetTableNode);
+                if("same-nodes".equals(targetTableNode)) {
+                    targetTableNode.append("\n" + getLineageNodeByColName.getTableDotFromNodeId(lineageRelation.getLineageNodeByTargetNodeId(), srcTableNodeList.get(0)) + "\n");
+                }
+                LOGGER.info("-------------------------------relation DOT: \n" + relationDot);
+            }
+
+        }
+        dotString = lineageMain.generateLineageDot(relationDot.toString(), srcTableNodeList.get(0).getDotString(), targetTableNode.toString());             //call lineage main to create the DOT
+        return dotString;
+    }
+
+    //fn() to get DOT through only Table name
+    public String dotGeneratorWithTable(String[] args) {
+        if(args.length !=1) {
+            LOGGER.info("One argument required: please Give Table name");
+            System.exit(-1);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this call to "exit" or ensure it is really required. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1147)